src/vast_tracker.js

```diff
@@ -536,7 +592,29 @@ export class VASTTracker extends EventEmitter {
       ) {
         variables['ASSETURI'] = this.creative.mediaFiles[0].fileURL;
       }
-      variables['CONTENTPLAYHEAD'] = this.progressFormatted();
+
+      const progress = this.progressFormatted();
+      // Do not set contentPlayHead if progress is NaN or already set
+      if (!variables['CONTENTPLAYHEAD'] && progress !== progress) {
+        //CONTENTPLAYHEAD @deprecated in VAST 4.1 replaced by ADPLAYHEAD & CONTENTPLAYHEAD
+        variables['CONTENTPLAYHEAD'] = progress;
+        variables['MEDIAPLAYHEAD'] = variables['ADPLAYHEAD'] =
+          variables['CONTENTPLAYHEAD'];
+      }
+    }
+    if (
+      this.creative &&
+      this.creative.universalAdId &&
+      this.creative.universalAdId.idRegistry &&
+      this.creative.universalAdId.value
+    ) {
+      variables['UNIVERSALADID'] = `${this.creative.universalAdId.idRegistry} ${
+        this.creative.universalAdId.value
+      }`;
```

Comment:
is this according to the specs? 🤔

Reply:
Yes, it's on p.97: `Format:  "registryID idvalue" - with a space separating registryID and value.`